### DISCUSSION TITLE
How to reference an external schema for a model

### DISCUSSION THREAD

**Discussion by booooh** [fastapi/fastapi#6217] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Foo(BaseModel):
    pass
```

### Description

I'm looking for a way to override the '$ref' field of a model so that I can reference an external schema. The schema is hosted on a different server, as the model is shared between multiple micro-services.

I have a pydantic model say `Foo` and I'd like to reference an external schema:
```python
class Foo(BaseModel):
    pass
```
And I would like it to reference the schema at https://example.com/api/v1/openapi.json#/components/schemas/Foo
Is there a way to customize the openapi documentation reference that URL instead of the local one?

### Operating System

Linux

### FastAPI Version

0.73

### Python Version

3.6.13

**Comment by booooh** [comment, marked as answer]
Answering myself:

1. It is a pydantic thing after all
2. You can override a pydantic Model's schema using the schema_extra property Config class for the Model:

```python
class Foo(BaseModel):

    class Config:
        def schema_extra(s):
            s.clear()
            s["$ref"] = "https://example.com/api/v1/openapi.json#/components/schemas/Foo"

```

**Reply by afandian** [reply]
Thanks for coming back to give the answer! I found this useful: https://stackoverflow.com/questions/71035845/example-in-schema-extra-is-ignored-by-pydantic-in-fastapi , especially the `json_schema_extra` comment.
